doc/page-cmakefunctions.h

```diff
@@ -0,0 +1,173 @@
+\param IMPORT_FOLDERS import interfaces from the folder paths
+
+<hr>
+
+
+\code
+function(facelift_generateQRC OUTPUT_FILE)
+\endcode
+Generates the qrc resource file from the given input folders. Function will look for js, qml, ttf , png, sci, json, astcz, tcsh, tesh, gsh, fsh files.
+\param OUTPUT_FILE  output qrc resource file
+\param INPUT_FOLDERS list of folders containing resources to be parsed
+
+<hr>
+
+\code
+function(facelift_add_library TARGET_NAME)
+\endcode
+
+\param TARGET_NAME name of the library
+\param LIBRARY_TYPE type of library shared static
+\param SOURCES sources files compiled to create a library
+\param HEADERS header files compiled to create a library
+\param LINK_LIBRARIES  external library names to be included
+\param SOURCES_GLOB_RECURSE list of cpp files for the plugin, could be provided as a wildcard expression
+\param HEADERS_GLOB_RECURSE list of h files for the plugin, could be provided as a wildcard expression
+\param EXPORT Create the directory for the installed headers
+\param INSTALL Install library
+
+<hr>
+
+
+\code
+function(facelift_add_executable TARGET_NAME)
+\endcode
+Add an executable to the project using the specified source files.
+\param TARGET_NAME name of the executable file
+
+<hr>
+
+\code
+function(facelift_add_test TEST_NAME)
+\endcode
+Adds the test with TEST_NAME as test case name
+\param TEST_NAME name of the test case
+\param SOURCES source cpp file 
+\param HEADERS header file
+\param LINK_LIBRARIES link library if needed
+
+<hr>
+
+
+
+\section private-functions Functions for FaceLift maintainers
+
+These functions are meant to be used by FaceLift maintainers, a.k.a by FaceLift internally.
+
+\code
+function(facelift_add_unity_files TARGET_NAME VAR_NAME)
+\endcode
+Generates unity files for targetname provided in the function
+\param TARGET_NAME target name of the file
+\param VAR_NAME  optional
+<hr>
+
+\code
+function(facelift_synchronize_folders FOLDER_SOURCE FOLDER_DESTINATION)
+\endcode
+Copy the content of FOLDER_SOURCE into FOLDER_DESTINATION, without overwriting files which already have the same content
+
+\param FOLDER_SOURCE  files to be copied from
+\param FOLDER_DESTINATION files to be copied to
+
+<hr>
+
+\code
+function(facelift_generate_code )
+\endcode
+Generates facelift code  from interface qface files 
+\param INTERFACE_DEFINITION_FOLDER folder where qface interface files are located
+\param OUTPUT_PATH output path of the generated code
+\param LIBRARY_NAME name of the library 
+\param IMPORT_FOLDERS path of import folders, if any external import dependency  is used
+
+<hr>
+
+
+\code
+function(facelift_export_project)
+\endcode
+This allows users of our package to use the "find_package(PACKAGE_NAME)" statement to retrieve the definitions needed to build against our package
+\param BUILD_FILES  all build files of the project
+\param INSTALLED_FILES all files user wants to install 
+
+<hr>
+
+
+\code
+function(facelift_append_to_file_if_not_existing FILE_PATH CONTENT)
```

Comment:
no public function

Reply:
its not public , i am putting these in private section